### PR TITLE
Ruby group capstone - Catalog of my things

### PR DESCRIPTION
**Pull request by ridaarif98**
> In this project, we have created a console app that will help you to keep a record of different types of things you won: books, music albums, movies, and games.
- Each class from the [ UML class diagram](https://github.com/microverseinc/curriculum-ruby/blob/main/group-capstone/images/catalog_of_my_things.png) are created in a separate .rb file.
-  All associations (1-to-many relationships) and aggregations (parent-child relationships) between classes are implemented.
- All properties visible in the diagram are defined and set up in the constructor method. _Exception: properties for the 1-to-many relationships are NOT  set in the constructor method. Instead, they have a custom setter method created._
- All methods visible in the diagram are implemented:
    - `add_item` method (in all classes that have that method)
        -  take an instance of the Item class as an input.
        - add the input item to the collection of items.
        - add self as a property of the item object (by using the correct setter from the item object).
    - `can_be_archived?()` in the Item class
        - return true if published_date is older than 10 years.
        - otherwise, it returns false.
    - `move_to_archive()` in the Item class
        - reuse `can_be_archived?()` method.
        - change the archived property to true if the result of the `can_be_archived?()` method is true.
        - do nothing if the result of the `can_be_archived?()` method is false.
    - `can_be_archived?()` in the Book class
        - override the method from the parent class.
        - returns true if parent's method returns true OR if cover_state equals to "bad".
        - otherwise, it returns false.
    - `can_be_archived?()` in the MusicAlbum class
        - override the method from the parent class.
        - returns true if parent's method returns true AND if on_spotify equals true.
        - otherwise, it returns false.
    - `can_be_archived?()` in the Movie class
        - override the method from the parent class.
        - returns true if the parent's method returns true OR if silent equals true.
        - otherwise, it returns false.
    - `can_be_archived?()` in the Game class
        - override the method from the parent class.
        - returns true if parent's method returns true AND if last_played_at is older than 2 years.
        - otherwise, it returns false.
- Add unit tests for all implemented methods.

#### User interface

- Create a `main.rb` file that serves as your console app entry-point.
- Our console app, at the start:
    - Present the user with a list of options to perform.
    - Let users choose an option.
    - If needed, ask for parameters for the option.
    - Have a way to quit the app.
- The following options are available:
    - List all books
    - List all music albums
    - List all movies
    - List of games
    - List all genres (e.g 'Comedy', 'Thriller')
    - List all labels (e.g. 'Gift', 'New')
    - List all authors (e.g. 'Stephen King')
    - List all sources (e.g. 'From a friend', 'Online shop')
    - Add a book
    - Add a music album
    - Add a movie
    - Add a game
 - All data is preserved by saving collections in `.json` files.

### Database
- Create a `schema.sql` file with tables that is analogical to the structure of the classes in our app. As we cannot implement inheritance in the database tables - add all properties and associations from the parent Item class as table columns to all tables based on the child classes.

### CODE REVIEW

File: app_menu.rb [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+      option = gets.chomp
+      break if option == '10'
+
+      handle_option option
+    end
+    save_books
+    save_labels
+    save_music
+    save_genre
+  end
```
Comment (EricMbouwe) [inline app_menu.rb:41]: - [ ] Let's make sure that the app does not crash if the `.json `file does not exist or is empty. Currently, that is the case after adding some data and then removing or deleting a json file.

Screenshot from 2021-11-25 15-17-39

**Review by EricMbouwe — CHANGES_REQUESTED** [review]
### Hello team,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlight!
- Good PR ✔️
- Good commits ✔️
- Good presentation of the project ✔️
- Good test coverage ✔️
- Good code organization and working app ✔️

# Required Changes ♻️ ⛔️

_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by od-c0d3r — APPROVED** [review]
Asslam Alaikum Team @ridaarif98 @julie-ify @Kalunge   ,😊

Hi👋🏽, Your project is complete! 

# Congratulations! 🎉 

Happy coding!👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ridaarif98** [comment]
@od-c0d3r  thank you 😃

**Pull request merged**